packages/botonic-plugin-google-analytics-v4/src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { PluginPostRequest, PluginPreRequest } from '@botonic/core'
+import axios from 'axios'
+
+import { BotSession, GA4Event, GA4Options } from './types'
+
+const defaultGetClientId = (session: BotSession) => session.user.id
+
+export default class BotonicPluginGoogleAnalytics4 {
+  private baseUrl = 'https://www.google-analytics.com/mp/collect'
+  private apiSecret: string
+  private measurementId: string
+  private getClientId: (session: BotSession) => string
+  private userId?: string
```

Comment:
I suggest to set the `userId` also as a function (`getUserId`) the same way it's done for `getClientId`, to allow the bot to decide where to obtain its value rather than fixing it when initializing the plugin.

Also, its value should be added in the constructor.

Reply:
added, thanks for the feedback 🙌